proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
     -> UnsafeMutablePointer<T>
 
-  func initialize<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func initializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func initialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
   // The `move` APIs deinitialize the memory at `from`.
-  func moveInitialize<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, forwardToCount: Int)
     -> UnsafeMutablePointer<T>
-  func moveInitializeBackward<T>(from: UnsafePointer<T>, count: Int)
+  func moveInitialize<T>(from: UnsafePointer<T>, backwardToCount: Int)
     -> UnsafeMutablePointer<T>
 
-  func load<T>(_: T.Type) -> T
-  func load<T>(_: T.Type, atByteOffset: Int) -> T
-  func load<T>(fromContiguous: T.Type, atIndex: Int) -> T
+  func load<T>(as: T.Type) -> T
+  func load<T>(fromByteOffset: Int, as: T.Type) -> T
+  func load<T>(fromContiguous: T.Type, at: Int) -> T
 
   // storeRaw performs bytewise writes, but proper alignment for `T` is still
   // required.
   // T must be a trivial type.
-  func storeRaw<T>(_: T.Type, with: T)
-  func storeRaw<T>(toContiguous: T.Type, atIndex: Int, with: T)
-  func storeRaw<T>(contiguous: T.Type, from: UnsafeRawPointer, count: Int)
-  func storeRawBackward<T>(
-    contiguous: T.Type, from: UnsafeRawPointer, count: Int)
+  func storeRaw<T>(_: T, as: T.Type)
+  func storeRaw<T>(_: T, toContiguous: T.Type, at: Int)
+  func storeRaw<T>(_: T, toByteOffset: Int)
+  func storeRaw<T>(
+    contiguous: T.Type, from: UnsafeRawPointer, forwardToCount: Int)
+  func storeRaw<T>(
+    contiguous: T.Type, from: UnsafeRawPointer, backwardToCount: Int)
```

Comment:
I realized I have not gotten enough feedback on this method:
 `fun storeRaw<T>(contiguous: T.Type, from: UnsafeRawPointer,
forwardToCount: Int)`

This is the `memcpy` API. It is very similar to bulk initialization:

`ptrToInt = rawPtr.initializeMemory(as: Int.self, from: srcAddress, count: n)`

But in this case, the memory does not need to be uninitialized, no
values are constructed, the memory is not bound, and no typed pointer
is returned.

For single value stores we have:
`rawPtr.store(x, asTrivial: Int.self)`

For bulk stores I think we should simply have:

`rawPtr.store(bytes: strideof(Int.self) * n, from: srcAddress)`

Proposed:
`func store<T>(bytes: Int, from: UnsafeRawPointer)`